stl/inc/algorithm

```diff
@@ -961,6 +941,340 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
+// FUNCTION TEMPLATE is_permutation
+template <class _FwdIt1, class _FwdIt2, class _Pr>
+_NODISCARD _CONSTEXPR20 bool _Is_permutation_unchecked(_FwdIt1 _First1, _FwdIt1 _Last1, _FwdIt2 _First2, _Pr _Pred) {
+    // test if [_First1, _Last1) == permuted [_First2, ...), using _Pred
+    for (; _First1 != _Last1; ++_First1, (void) ++_First2) { // trim matching prefix
+        if (!_Pred(*_First1, *_First2)) {
+            // found first inequality, check match counts in suffix
+            //
+            // narrowing _Iter_diff_t<_FwdIt1> to _Iter_diff_t<_FwdIt2> is OK because if the second range is shorter

```

Comment:
Playing @StephanTLavavej here. Should that be a proper sentence with period at the end and uppercase Narrowing